src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeCustomAttributeData.cs

```diff
@@ -273,6 +273,13 @@ private RuntimeCustomAttributeData(RuntimeModule scope, MetadataToken caCtorToke
             m_scope = scope;
             m_ctor = (RuntimeConstructorInfo)RuntimeType.GetMethodBase(scope, caCtorToken)!;
 
+            if (m_ctor!.DeclaringType!.IsGenericType)
+            {
+                MetadataImport metadataScope = scope.MetadataImport;
+                var attributeType = scope.ResolveType(metadataScope.GetParentToken(caCtorToken), null, null)!;
+                m_ctor = (scope.ResolveMethod(caCtorToken, attributeType.GenericTypeArguments, null)!.MethodHandle.GetMethodInfo() as RuntimeConstructorInfo)!;
```

Comment:
Would it be better to just do a C-style case here if `null` isn't possible? I am assuming that based on the trailing `!`.